### DISCUSSION TITLE
How to set delete_repo scope only for specific repositories?

### DISCUSSION THREAD

**Discussion by lyqht** [octokit/octokit.js#2230] [Q&A] [answered]
From what i understand from the [Oauth docs](https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps), the `delete_repo` scope grants access to delete **all** adminable repositories.

Is there a way to restrict to only specific repositories?

Thank you!

**Comment by gr2m** [comment, marked as answer]
Unfortunately no, OAuth Scopes are global. 

An alternative would be to register a GitHub App with `administration:write` permission, install the app on the repositories you want, and then delete the repositories using the app's credentials, see the [README example](https://github.com/octokit/octokit.js#github-app) (instead of creating a dispatch event, delete the repository)
